name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesSelectionDialog.java

```diff
@@ -138,18 +139,48 @@ public void setElements(List<DataSeries> elements)
             Node child = new Node(series.getSearchLabel());
             child.dataSeries = series;
 
-            Node parent = type2node.computeIfAbsent(map(series.getType()), Node::new);
+            Node parent = type2node.computeIfAbsent(map(series), Node::new);
 
-            if (series.getGroup() != null)
+            if (series.getGroups() != null)
```

Comment:
You have to keep in mind, that taxonomies can have an arbitrary depth.
That depth used to be reflected in the dialog as well.

See the screenshot from the current version:
<img width="269" alt="Bildschirmfoto 2024-01-26 um 21 11 40" src="https://github.com/portfolio-performance/portfolio/assets/587976/970777cf-e1d6-4446-b7b7-ad254d00a0e6">

This pull requests flatten the classifications - which makes it hard to select (particularly because the name can repeat itself). This is how it looks with your pull request:

<img width="303" alt="Bildschirmfoto 2024-01-26 um 21 11 30" src="https://github.com/portfolio-performance/portfolio/assets/587976/4b9479a9-b883-4902-914a-ccb05643841b">

I understand your reasoning.
If a node ist about a classification with the types of data series (total, invested capital, etc.) then it is hard for the same node to _also_ have the sub-nodes. But flattening the classifications does make it much harder I think.

(let's discuss the UI interaction in the comments of the pull request)

Reply:
Okay, I realize it never was the case. The path was just added in order to separate the items.

<img width="376" alt="Bildschirmfoto 2024-01-26 um 22 17 33" src="https://github.com/portfolio-performance/portfolio/assets/587976/4ec2fa24-6be6-48f4-a2c2-b26fe7aa207a">

At minimum we still need the path.

Reply:
<img width="231" alt="image" src="https://github.com/portfolio-performance/portfolio/assets/11842979/a4d8c132-4b58-459a-abbc-8a3b3a2c3fc2">
I've added that in 56758c6bd